ci/jenkins/Jenkins_steps.groovy

```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+            utils.pack_lib('r-artifacts', 'docs/_build/r-artifacts.tgz', false)
+          }
+        }
+      }
+    }]
+}
+
+
+def docs_scala() {
+    return ['Scala Docs': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.unpack_and_init('libmxnet', mx_lib, false)
+            utils.docker_run('ubuntu_cpu_scala', 'build_scala_docs', false)
+            utils.pack_lib('scala-artifacts', 'docs/_build/scala-artifacts.tgz', false)
+          }
+        }
+      }
+    }]
+}
+
+
+def docs_java() {
+    return ['Java Docs': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.unpack_and_init('libmxnet', mx_lib, false)
+            utils.docker_run('ubuntu_cpu_scala', 'build_java_docs', false)
+            utils.pack_lib('java-artifacts', 'docs/_build/java-artifacts.tgz', false)
+          }
+        }
+      }
+    }]
+}
+
+
+def docs_clojure() {
+    return ['Clojure Docs': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.unpack_and_init('libmxnet', mx_lib, false)
+            utils.docker_run('ubuntu_cpu_scala', 'build_clojure_docs', false)
+            utils.pack_lib('clojure-artifacts', 'docs/_build/clojure-artifacts.tgz', false)
+          }
+        }
+      }
+    }]
+}
+
+
+def docs_jekyll() {
+    return ['Main Jekyll Website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.init_git()
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_jekyll_docs', false)
+            utils.pack_lib('jekyll-artifacts', 'docs/_build/jekyll-artifacts.tgz', false)
+          }
+        }
+      }
+    }]
+}
+
+
+// This is for publishing the full website
+def docs_prepare() {
+    return ['Prepare for publication of the full website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.init_git()
+
+            unstash 'jekyll-artifacts'
+            unstash 'c-artifacts'
+            unstash 'python-artifacts'
+            unstash 'r-artifacts'
+            unstash 'julia-artifacts'
+            unstash 'scala-artifacts'
+            unstash 'java-artifacts'
+            unstash 'clojure-artifacts'
+
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_docs', false)
+
+            // only stash if we're going to unstash later
+            // utils.pack_lib('full_website', 'docs/_build/full_website.tgz', false)
+
+            // archive so the publish pipeline can access the artifact
+            archiveArtifacts 'docs/_build/full_website.tgz'
+          }
+        }
+      }
+    }]
+}
+
+
+// This is for the website and the Python API only
+def docs_prepare_python_only() {
```

Comment:
Thanks!

It'd be ideal to have some more description in the comments on these methods as it's unclear out of context how things would work.    Also the distinction b/w jekyll and non-jekyll will be lost over time, so a comment will help.